src/clients/slaos.py

```diff
@@ -0,0 +1,38 @@
+from datetime import datetime
+import httpx
+
+from src.config.models.offset import SLIIndexedDocument
+from src.config.models.output import RatedOutputConfig
+
+
+class SlaosClient:
+    def __init__(self, config: RatedOutputConfig):
+        self.config = config
+        self.client = httpx.Client()
+
+    @property
+    def full_ingest_url(self) -> str:
+        base_url = self.config.ingestion_url
+        ingestion_id = self.config.ingestion_id
+        ingestion_key = self.config.ingestion_key
+
+        return f"{base_url}/{ingestion_id}/{ingestion_key}"
+
+    def get_latest_ingest_timestamp(self, datastream_key: str) -> datetime | None:
+        url = f"{self.full_ingest_url}/indexed-slis"
+        params = {
+            "key": datastream_key,
+            "limit": "1",
+        }
+        headers = {
+            "Content-Type": "application/json",
+        }
+        # TODO: handle all manner of exceptions in this block
+        response = self.client.get(url, params=params, headers=headers)
+        response.raise_for_status()
```

Comment:
@samtin0x Will there be a need to handle this more elegantly? We may get errors from misconfigurations or even from the API being unavailable. Are these errors okay or do we fall back to the start time that's in the config?

Reply:
> I'd suggest if there's another option where we only fetch it once at the beginning. So we fetch it and don't make more network requests and then if instance breaks or something, on restart we fetch it from the API again

Yes, that's how this is intended to work. That doesn't mean there will never be errors.